src/V3LinkDot.cpp

```diff
@@ -2762,7 +2762,13 @@ class LinkDotResolveVisitor final : public VNVisitor {
                 if (AstVar* const varp = VN_CAST(foundp->nodep(), Var)) {
                     if (varp->isParam() || varp->isGenVar()) {
                         // Attach found Text reference to PatMember
-                        nodep->varrefp(new AstVarRef{nodep->fileline(), varp, VAccess::READ});
+                        if (foundp->imported()) {
+                            nodep->varrefp(new AstVarRef{nodep->fileline(),
+                                                         foundp->classOrPackagep(), varp,
+                                                         VAccess::READ});
+                        } else {
+                            nodep->varrefp(new AstVarRef{nodep->fileline(), varp, VAccess::READ});
+                        }
```

Comment:
Thanks for debugging, one simplification (remember to reindent).

```suggestion
                            nodep->varrefp(new AstVarRef{nodep->fileline(),
                                                         foundp->imported() ? foundp->classOrPackagep() : nullptr, varp,
                                                         VAccess::READ});
```